sycl/doc/design/OffloadDesign.md

```diff
@@ -0,0 +1,396 @@
+# Implementation design for offloading model
```

Comment:
Can we reword the Topic to be more specific to what we cover here? "Updated offloading compilation flow using clang-linker-wrapper'? Maybe?